### PR TITLE
Make EditableText respect MediaQuery.boldTextOf

### LINKED ISSUE #122750: `EditableText` does not respect `MediaQuery.boldTextOf(context)`.
Cause of b/250894028

Repro steps: launch any Flutter app with editable text, set `Bold Text` to true in a11y settings on iOS, observe fonts in editable text do not get bolded (whereas other text does).

@chunhtai fyi

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/122750

### CODE REVIEW

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 1]
```diff
@@ -4636,6 +4636,10 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   /// By default makes text in composing range appear as underlined.
   /// Descendants can override this method to customize appearance of text.
   TextSpan buildTextSpan() {
+    final TextStyle style = MediaQuery.boldTextOf(context)
```
Comment (dnfield) [inline packages/flutter/lib/src/widgets/editable_text.dart:4639]: There are other codepaths in this file that talk to the platform channel about text style. I'm not clear on whether those need to be updated as well. @LongCatIsLooong or @mdebbar may know

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/editable_text.dart:4639]: It looks like `SelectableText` is already doing this:

https://github.com/flutter/flutter/blob/7ce6ddabab5cee29d00fc20e2b8e4d5c99387bce/packages/flutter/lib/src/material/selectable_text.dart#L680-L682

So maybe TextField/CupertinoTextField are better places for this? Usually styles are resolved there iirc.

Comment (dnfield) [reply packages/flutter/lib/src/widgets/editable_text.dart:4639]: Yes, `SelectableText` does this, but `EditableText` does not.

If we do it higher up in the stack, couldn't some custom text editing class end up missing out? This seems like the right peer of `SelectableText` to me.

Comment (dnfield) [reply packages/flutter/lib/src/widgets/editable_text.dart:4639]: Ohh, I see what you're saying now.

Why not do this on `EditableText` and remove it from `SelectableText`?

Comment (dnfield) [reply packages/flutter/lib/src/widgets/editable_text.dart:4639]: (I removed the call from SelectableText - I had seen it before but missed the part where SelectableText just returns an `EditableText` anyway)

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/editable_text.dart:4639]: Ah ok. I was worried about not being able to opt-out but people can just add another MediaQuery. But I think `TextInputConnection.setTextStyle` needs to be updated as well. Also is this supposed to affect things like suffix text or placeholder text in `InputDecorator`?

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/editable_text.dart:4639]: Ah ok those are `Text` widgets. But I think `TextInputConnection.setTextStyle` still needs to be updated with the right font weight if `boldText` is on.

Comment (dnfield) [reply packages/flutter/lib/src/widgets/editable_text.dart:4639]: SG, I will do that and update the tests.

Comment (dnfield) [reply packages/flutter/lib/src/widgets/editable_text.dart:4639]: Done

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 2]
```diff
@@ -4631,11 +4631,22 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     );
   }
 
+  /// Checks the [MediaQuery.boldTextOf] on the [context].
+  ///
+  /// This makes sure that accessibility settings related to bold text are
+  /// respected.
+  TextStyle updateStyleForBold() {
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/editable_text.dart:4638]: should the update function be called in didChangeDependencies and cache the value?

Comment (chunhtai) [reply packages/flutter/lib/src/widgets/editable_text.dart:4638]: also this should probably be private

**Review by mdebbar — APPROVED** [review]
As long as we continue to send style updates to the engine, this looks good to me from web point of view!

File: packages/flutter/lib/src/widgets/editable_text.dart (hunk 2 above)
Comment (dnfield) [reply packages/flutter/lib/src/widgets/editable_text.dart:4638]: Deleted this method and just inlined it into `didChangeDependencies` and `didUpdateWidget`.

**Review by LongCatIsLooong — APPROVED** [review]

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 3]
```diff
@@ -2589,6 +2590,10 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   void didChangeDependencies() {
     super.didChangeDependencies();
 
+    _style = MediaQuery.boldTextOf(context)
+        ? widget.style.merge(const TextStyle(fontWeight: FontWeight.bold))
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/editable_text.dart:2594]: Sorry I didn't notice this also uses widget.style, we may also need to do this in didUpdateWidget.

Comment (chunhtai) [reply packages/flutter/lib/src/widgets/editable_text.dart:2594]: oh nvm, you updated it

**Review by chunhtai — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/editable_text.dart (hunk 3 above)
Comment (dnfield) [reply packages/flutter/lib/src/widgets/editable_text.dart:2594]: Yeah, thankfully we have test coverage :)

**Pull request merged**